### PR TITLE
Add permission check when creating PR

### PR DESCRIPTION
**Pull request by yp05327**
user should be a collaborator of the base repo to create a PR

### CODE REVIEW

**Comment by yp05327** [comment]
Some tests using organization user as the PR poster.....It will take lots of time to fix them. 😕

**Comment by yp05327** [comment]
@Zettat123 
The poster is org user in `TestPullRequestTargetEvent`, can it be converted into user4?

**Comment by Zettat123** [comment]
> @Zettat123 The poster is org user in `TestPullRequestTargetEvent`, can it be converted into user4?

My fault. It should be converted to another non-org user.

File: services/pull/pull.go [hunk 1]
```diff
@@ -48,6 +48,15 @@ func NewPullRequest(ctx context.Context, repo *repo_model.Repository, issue *iss
 		return user_model.ErrBlockedUser
 	}
 
+	// user should be a collaborator or a member of the organization
+	if !issue.Poster.IsAdmin {
```
Comment (lunny) [inline services/pull/pull.go:54]: I don't think the logic is right. https://github.com/go-gitea/gitea/pull/31033#issuecomment-2124966661

**Comment by yp05327** [comment]
> If he creates a PR from a forked repository, he should be a collaborator of the head repository.

This is not the Github's logic

**Comment by yp05327** [comment]
See: https://github.com/yp05327/test/pull/7

yp05328 is a collaborator of `yp05327/test`
but yp05328 is not a member of `yp05327test`

**Review by lunny — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Comment by yp05327** [comment]
Conflict fixed.

**Pull request merged**